### PR TITLE
refactor(client): make all fixtures use default options

### PR DESCRIPTION
**Pull request by leafty**
Details:

* Fixtures now have sensible defaults and callers can provide only the options they need.
* Added types for most common types of fixtures

### CODE REVIEW

**Review by ciyer — CHANGES_REQUESTED** [review]
This is a good change to unify the structure of the fixtures. 🎉 I have a few requests for field name changes to clarify the semantics of the field, and a suggestion for a refactoring.

**Review by ciyer — CHANGES_REQUESTED** [review]
Also, the `dashboard.spec.ts` fails, even locally, so I suspect an error was introduced into that test that needs to be fixed.

File: tests/cypress/support/renkulab-fixtures/projects.ts [hunk 1]
```diff
@@ -17,488 +17,721 @@
...
-      const fixture = this.useMockedData
-        ? { fixture: `projects/no-project.json`, statusCode: 404 }
-        : undefined;
+    errorProject(args?: ErrorProjectArgs) {
+      const { branches, project } = Cypress._.defaultsDeep({}, args, {
+        branches: { name: "getErrorProjectBranches", statusCode: 404 },
+        project: {
+          fixture: "projects/no-project.json",
+          name: "getErrorProject",
+          path: "",
```
[the remark this answers is not in the document]
Comment (leafty) [reply tests/cypress/support/renkulab-fixtures/projects.ts:188]: Done in [e1b88f9](https://github.com/SwissDataScienceCenter/renku-ui/pull/2864/commits/e1b88f991d0c315bcf8c3d77173fd1bc4b6bc4ad).

File: tests/cypress/support/renkulab-fixtures/projects.ts [hunk 2]
```diff
@@ -17,488 +17,721 @@
...
-        `/ui-server/api/projects/${encodeURIComponent(
-          path
-        )}?statistics=${statistics}&doNotTrack=*`,
-        fixture
-      ).as(name);
+    project(args?: Partial<ProjectArgs>) {
+      const {
+        fixture = "projects/project.json",
+        name = "getProject",
+        path = "",
```
[the remark this answers is not in the document]
Comment (leafty) [reply tests/cypress/support/renkulab-fixtures/projects.ts]: Done in [e1b88f9](https://github.com/SwissDataScienceCenter/renku-ui/pull/2864/commits/e1b88f991d0c315bcf8c3d77173fd1bc4b6bc4ad).

File: tests/cypress/support/renkulab-fixtures/datasets.ts [hunk 3]
```diff
@@ -35,184 +36,237 @@ function toLegacyIdentifier(datasetId) {
...
-    ) {
-      const fixture = this.useMockedData ? { fixture: resultFile } : undefined;
-      cy.intercept(`/ui-server/api/kg/projects/${path}/datasets`, fixture).as(
-        name
-      );
+    projectKGDatasetList(args?: ProjectKGDatasetListArgs) {
+      const {
+        fixture = "datasets/project-dataset-kg-list.json",
+        name = "datasetKGList",
+        path = "",
```
[the remark this answers is not in the document]
Comment (leafty) [reply tests/cypress/support/renkulab-fixtures/datasets.ts:78]: Done in [e1b88f9](https://github.com/SwissDataScienceCenter/renku-ui/pull/2864/commits/e1b88f991d0c315bcf8c3d77173fd1bc4b6bc4ad).

[the remark this answers is not in the document]
Comment (ciyer) [reply tests/cypress/support/renkulab-fixtures/datasets.ts:78]: It looks like this one was missed. That commit only changes `tests/cypress/support/renkulab-fixtures/projects.ts`, not this file.

File: tests/cypress/support/renkulab-fixtures/global.ts [hunk 4]
```diff
@@ -17,116 +17,152 @@
...
-      cy.intercept("https://*.statuspage.io/api/v2/summary.jsonr", {
-        statusCode: 500,
-        body: {},
-      }).as(name);
+    statuspageDown(args?: NameOnlyFixture) {
+      const { name = "getStatuspageInfo" } = args ?? {};
+      const response = { body: {}, statusCode: 500 };
+      cy.intercept(
+        "GET",
+        "https://*.statuspage.io/api/v2/summary.jsonr",
```
Comment (ciyer) [inline tests/cypress/support/renkulab-fixtures/global.ts:68]: This is a mistake introduced by my PR, but I will fix it in #2871

**Review by ciyer — CHANGES_REQUESTED** [review]
One of the changes requested in an earlier review was. missed, but this otherwise looks good and is very close to approval.

**Review by ciyer — APPROVED** [review]
LGTM!

**Pull request merged**